### PR TITLE
BI-1650 - Store and fetch genotype data

### PR DESCRIPTION
**Pull request by timparsons**
# Description
**Story:** https://breedinginsight.atlassian.net/browse/BI-1650

- Removing the requirement that a DB be created from the Gigwa UI
- Change how processIDs are created for importing VCF files to not use the auth token as the processID
- Updates to the Dockerfile to support multi-architecture builds
- Updates to the docker-compose.yml file to allow for variables to be defined in an `.env` file

# Dependencies

# Testing
- Verify that Gigwa starts up
- Verify that submitting genotype data to Gigwa from the REST endpoints (not the UI) allows the uploaded data to persist longer than 1 day

# Checklist:

- [x] I have performed a self-review of my own code
- [x] I have tested my code and ensured it meets the acceptance criteria of the story
- [ ] I have create/modified unit tests to cover this change
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to documentation

### CODE REVIEW

**Review by dmeidlin — APPROVED** [review]

File: src/main/java/fr/cirad/web/controller/gigwa/GigwaRestController.java [hunk 1]
```diff
@@ -1807,7 +1802,7 @@ public void run() {
 						try {							
 							Integer newProjId = null;
 							if (fBrapiImport)
-								newProjId = new BrapiImport(token).importToMongo(sNormalizedModule, sProject, sRun, sTechnology == null ? "" : sTechnology, dataUri1.trim(), sBrapiStudyDbId, sBrapiMapDbId, sBrapiToken,  Boolean.TRUE.equals(fClearProjectData) ? 1 : 0);
+								newProjId = new BrapiImport(processId).importToMongo(sNormalizedModule, sProject, sRun, sTechnology == null ? "" : sTechnology, dataUri1.trim(), sBrapiStudyDbId, sBrapiMapDbId, sBrapiToken,  Boolean.TRUE.equals(fClearProjectData) ? 1 : 0);
```
Comment (nickpalladino) [inline src/main/java/fr/cirad/web/controller/gigwa/GigwaRestController.java]: Is this process id change going to be maintained separately from upstream code? Any plans to see if some sort of configurable option could be implemented?

Comment (timparsons) [reply src/main/java/fr/cirad/web/controller/gigwa/GigwaRestController.java]: @GuilhemSempere updated Gigwa to do this in the upstream repo.  I've rebased to this, so now it is in-line.

File: src/main/java/fr/cirad/web/controller/gigwa/GigwaRestController.java [hunk 2]
```diff
@@ -31,23 +31,9 @@
...
-import java.util.LinkedHashSet;
-import java.util.List;
-import java.util.Map;
-import java.util.Scanner;
-import java.util.Timer;
-import java.util.TimerTask;
-import java.util.TreeMap;
-import java.util.TreeSet;
-import java.util.UUID;
+import java.text.ParseException;
```
Comment (nickpalladino) [inline src/main/java/fr/cirad/web/controller/gigwa/GigwaRestController.java]: Are these added imports needed?

Comment (timparsons) [reply src/main/java/fr/cirad/web/controller/gigwa/GigwaRestController.java]: Reverted

**Review by nickpalladino — APPROVED** [review]

**Review by davedrp — APPROVED** [review]

**Pull request merged**